### PR TITLE
Basic Readme updates

### PR DESCRIPTION
**Pull request by prestwich**
Adds additional sections to the readme to cover endianness and general Request architecture.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,7 +41,65 @@ This model provides large gas savings compared to previous relay designs (TODO:
...
 
-### Project Notes
+### A Note on Endianness
+
+Bitcoin internally uses little-endian representations of integers and digests.
+Block explorers and most user-facing applications use the more common
+big-endian representation. To minimize order swaps and prevent confusion, all
+our tooling uses the LE representation exclusively. If using the JS, rust,
+golang, or python tooling in [bitcoin-spv](http://bitcoin-spv.com), everything
+will Just Work. If writing custom software using data from block explorers,
```
Comment (ErinHales) [inline README.md:51]: Lower case "will just work"

File: README.md [hunk 2]
```diff
@@ -41,7 +41,65 @@ This model provides large gas savings compared to previous relay designs (TODO:
...
+full nodes, or other data sources, make sure digests are LE before submitting
+to the relay.
```
Comment (ErinHales) [inline README.md:53]: Do we have a function for reversing endianness in Solidity?  We could mention that here.

File: README.md [hunk 3]
```diff
@@ -41,7 +41,65 @@ This model provides large gas savings compared to previous relay designs (TODO:
...
+
+### Requests and Proofs
+
+The Relay implementations here have a SPV request system built in. This allows
```
Comment (ErinHales) [inline README.md:57]: an SPV

File: README.md [hunk 4]
```diff
@@ -41,7 +41,65 @@ This model provides large gas savings compared to previous relay designs (TODO:
...
+a stream of transactions that meet the filter's specifications.
+
+**Note**: Due to solidity constraints, this filter system is unrelated to
+existing Bitcoin filtering systems (e.g. BIP37 & BIP157) In the future,
+the filter system may be upgraded to support more complex transaction
+descriptions.
+
+**Important**: All requests may be filled more than once. The handler should
+expect this situation. Setting a `spends` filter is NOT sufficient to prevent
+this, as long reorgs may cause a UTXO to be spend multiple times.
```
Comment (ErinHales) [inline README.md:100]: Add details for what WOULD be sufficient?

File: README.md (hunk 1 above)
Comment (prestwich) [reply README.md:51]: but I like the emphasis caps!

File: README.md (hunk 2 above)
Comment (prestwich) [reply README.md:53]: they _should_ never need the solidity functions

File: README.md (hunk 1 above)
Comment (ErinHales) [reply README.md:51]: In that case, revise to JUST DO IT ✔️ 😃

**Pull request merged**